src/app/(proper_react)/redesign/(public)/SignUpForm.tsx

```diff
@@ -44,16 +47,25 @@ export const SignUpForm = (props: Props) => {
           "landing-all-hero-emailform-input-placeholder",
         )}
       />
-      <Button type="submit" variant="primary" wide>
+      <Button
+        type="submit"
+        variant="primary"
+        wide
+        className={props.isHero ? styles.isHero : ""}
+      >
         {l10n.getString("landing-all-hero-emailform-submit-label")}
       </Button>
-      <label htmlFor={emailInputId}>
-        {l10n.getString(
-          props.eligibleForPremium
-            ? "landing-premium-hero-emailform-input-label"
-            : "landing-all-hero-emailform-input-label",
-        )}
-      </label>
+      {props.isHero ? (
```

Comment:
[All inputs need a label](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#accessibility_concerns):

> When including inputs, it is an accessibility requirement to add labels alongside. This is needed so those who use assistive technologies can tell what the input is for.

Ideally, they're also visible, to provide a larger click target. But since the design doesn't include them, and the input fields are pretty big anyway, we can probably just wrap them into a `<VisuallyHidden>` if it's not in the hero.

Reply:
Huh interesting, indeed I would expect axe to catch that. So I did some digging, and [it does have a rule for this](https://dequeuniversity.com/rules/axe/4.7/label), but apparently it says:

>  Lastly a placeholder attribute may be used to give text inputs an accessible name. This is _not_ a recommended solution as the visual label (the placeholder text) will be removed once the user enters text into the input, causing them to not know what the input is for.

So the `placeholder` caused the test not to fail, but it's still recommended to add an actual label.